cmd/cluster/pullsecret.go

```diff
@@ -0,0 +1,90 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func newCmdPullSecret(kubeCli client.Client, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	return &cobra.Command{
+		Use:               "pull-secret",
+		Short:             "Checks if the pull secret email matches the owner email",
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ValidatePullSecret(args[0], kubeCli, flags))
+		},
+	}
+}
+
+func ValidatePullSecret(clusterID string, kubeCli client.Client, flags *genericclioptions.ConfigFlags) error {
+	ocm := utils.CreateConnection()
+	defer func() {
+		if ocmCloseErr := ocm.Close(); ocmCloseErr != nil {
+			fmt.Printf("Cannot close the ocm (possible memory leak): %q", ocmCloseErr)
+		}
+	}()
+
+	fmt.Println("Checking if pull secret email matches user email")
+
+	flags.Impersonate = pointer.StringPtr("backplane-cluster-admin")
```

Comment:
Correct me if I'm wrong, but I don't believe this ever used? If not, can we remove it?

Reply:
I stand corrected: it looks like this is the same flagset used by the kubeCli client.Client object [provided by the root cmd](https://github.com/openshift/osdctl/blob/master/cmd/cmd.go#L88). So we're retroactively impersonating backplane-cluster-admin after the client has been built, but, in my opinion, it's not immediately evident that's the case.

Could we maybe add a comment indicating as much for future reference?

Reply:
It also appears that the `pointer.StringPtr` function is [deprecated](https://pkg.go.dev/k8s.io/utils/pointer#StringPtr).

Reply:
I was typing a response to your first query but I see you've figured it out 😄 Yes, that flag is necessary to impersonate backplane-cluster-admin for our use cases here.

As for `StringPtr` being deprecated - good catch! I had copied this from `/cmd/account/generate-secret.go` and never went back to clean up. I'll do that now.

Reply:
Yeah, sorry about the initial confusion. Took my brain a minute to recall how the k8s client works :slightly_smiling_face: